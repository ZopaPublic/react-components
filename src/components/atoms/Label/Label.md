### Summary

Use `<Label />` to render a native HTML `<label />`.

### Example

```jsx
import { Label } from '@zopauk/react-components';

<Label htmlFor="username">Username</Label>;
```
