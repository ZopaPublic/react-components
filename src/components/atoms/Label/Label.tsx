import React from 'react';
import styled from 'styled-components';
import Text from '../Text/Text';
import { colors } from '../../../constants/colors';

export interface ILabelProps extends React.LabelHTMLAttributes<HTMLLabelElement> {}

const StyledLabel = styled(Text)`
  display: block;
`;

const Label = (props: ILabelProps) => (
  <StyledLabel {...props} weight="semibold" forwardedAs="label" color={colors.neutral.dark} size="base" />
);

export default Label;
