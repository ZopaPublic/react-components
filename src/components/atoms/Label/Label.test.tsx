import React from 'react';
import { render } from '@testing-library/react';
import Label from './Label';

describe('<Label />', () => {
  it('renders the component with props', () => {
    const { container } = render(<Label htmlFor="username">Username:</Label>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
