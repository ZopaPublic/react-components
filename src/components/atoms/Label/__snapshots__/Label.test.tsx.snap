// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> renders the component with props 1`] = `
.c0 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #0D0A38;
  line-height: 1.6;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-weight: 600;
  font-size: 16px;
}

.c1 {
  display: block;
}

<label
  class="c0 c1"
  color="#0D0A38"
  for="username"
>
  Username:
</label>
`;
